### ISSUE TITLE
Confusion on official cli binary name

### ISSUE THREAD

**Issue by jay-babu** [editorconfig-checker/editorconfig-checker#232]
Hello,

When this tool is installed, should the binary/cli tool be called `editorconfig-checker` or `ec`. Typically, shorthand names like `ec` are reserved for system tools such as `ls`, `cat`, etc.

When running

`go install github.com/editorconfig-checker/editorconfig-checker/cmd/editorconfig-checker@latest`, the binary name is `editorconfig-checker`. However, on the releases page, the binary is `ec`. This has sparked some discussions across a few plugins using this plugin:

https://github.com/jose-elias-alvarez/null-ls.nvim/pull/1282

https://github.com/jay-babu/mason-null-ls.nvim/issues/37

https://github.com/williamboman/mason.nvim/pull/722

What should these plugins assume the binary to be called by default: `editorconfig-checker` or `ec`?

**Comment by jay-babu** [comment]
Note: I am a fan of `editorconfig-checker`, but don't let that change what it is supposed to be.

**Comment by jay-babu** [comment]
@chalkygames123, would you be able to provide some guidance?

**Comment by chalkygames123** [comment]
Nah, I'm not a maintainer of this project. But I'm a fan of `editorconfig-checker` too.

**Comment by jay-babu** [comment]
@mstruebing, are you a maintainer? Would you be able to weigh in on this?

**Comment by mstruebing** [comment]
So, my intention to add `ec` as a possible binary name was just because it is shorter and I'm lazy!
I did not really know about the convention of system tools.

Do you have a strong opinion if it should rather be ec or editorconfig-checker?

@chalkygames123 @jay-babu 

I currently think it should be `editorconfig-checker` but I let your opinion count more than mine.

**Comment by jay-babu** [comment]
I would side with `editorconfig-checker`, based on the above convention. People usually alias if they want something shorter. It's a 1 line change for them.

**Comment by chalkygames123** [comment]
Having an alias binary is okay, but having **only** that one is not a good idea, IMO.

**Comment by jay-babu** [comment]
@mstruebing, can we close this issue with the acceptance of `editorconfig-checker` as the resolution?

**Comment by jemag** [comment]
If truly the confusion is to be removed, then I think it would be important to standardize it across the board.

Currently the `pacman` install and the `docker` image use `ec` and the README is also consistently referring to `ec`. For those reasons, I actually thought that the official binary name was `ec`.

 If `editorconfig-checker` is to be adopted instead, then it would be important to change all of those.

**Comment by janedbal** [comment]
I really feel that such huge BC break is not worth it. **Is there is real issue except confusions**?


> Typically, shorthand names like ec are reserved for system tools such as ls, cat, etc.

Can you support this claim by some standard saying that?

**Comment by robmoss2k** [comment]
This issue has been stale for months pending the above question, so I asked ChatGPT/o1. I think we should close this issue:

There is no POSIX, LSB, or other “official” standard that explicitly reserves short command names (such as `ls` or `cat`) for system binaries. Historically, many core Unix utilities simply had short names for ease of typing and to reflect the limited storage and memory constraints on older systems. Over time, these abbreviations became part of the Unix tradition, and POSIX does specify that certain utilities (e.g., `ls`, `cat`, `cd`, `cp`, etc.) must exist and behave in a particular way. However, that does not amount to a formal rule that **no** other program can use a short name.

In other words:

- **POSIX** and the Single UNIX Specification do require certain standard utilities to exist under specific names, but nowhere do they say that “short names are off-limits” to other software.  
- **Filesystem Hierarchy Standards (FHS)** and related guidelines discuss directory layout and the conventional locations of system binaries, but they likewise do not forbid non-system programs from having a short name.  
- The short, mnemonic names we commonly see (e.g., `ls`, `cp`, `mv`) became conventions rather than formal restrictions.  

Although there is no rule preventing you from creating or distributing a tool with a short name, it is **strongly discouraged** to ship an application that conflicts with a commonly used system command (e.g., your own `/usr/bin/ls`) because it causes confusion and breaks user expectations. But that caution is a best practice, not a requirement in a formal standard.

**Comment by klaernie** [comment]
In my opinion this breaking change is worth it - no matter if another util exists currently or if the standard is only of perceived nature.

We currently have a pull request open for changing the binary name, and have already started shipping releases that provide archives with the new binary name. Similarly the docker images contain the binary with both names.

So the issue is not stale, but currently being worked on, with the correct pace to reflect the impact of the change.
